### ISSUE TITLE
Name of async recursive function is stripped away

### ISSUE THREAD

**Issue by honzabrecka** [palmerhq/tsdx#172]
### Current Behavior

Following code:

```ts
export function pipeP(...fns: Function[]) {
  return (value: any) =>
    new Promise((resolve, reject) => {
      ;(async function run([f, ...fns], value: any) {
        try {
          if (f === undefined) resolve(value)
          else run(fns, await f(value))
        } catch (e) {
          reject(e)
        }
      })(fns, value)
    })
}
```

 is being compiled to:

```js
function pipeP() {
  for (var _len = arguments.length, fns = new Array(_len), _key = 0; _key < _len; _key++) {
    fns[_key] = arguments[_key];
  }

  return function (value) {
    return new Promise(function (resolve, reject) {

// function run - its name is stripped
      (function (_ref, value) {
        var f = _ref[0],
            fns = _ref.slice(1);

        try {
          return _catch(function () {
            var _temp = function () {
              if (f === undefined) resolve(value);else return Promise.resolve(f(value)).then(function (_f) {
// calling run fails, because theres's no run defined
                run(fns, _f);
              });
            }();

            if (_temp && _temp.then) return _temp.then(function () {});
          }, function (e) {
            reject(e);
          });
        } catch (e) {
          Promise.reject(e);
        }
      })(fns, value);
    });
  };
}
```

Fails with `run is not defined` - see comments above for explanation.

### Expected behavior

Does not strip function's name

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | 0.7.2
| TypeScript       | 3.5.3

**Comment by honzabrecka** [comment]
Causd by babel-plugin-transform-async-to-promises babel plugin, reported here https://github.com/rpetrich/babel-plugin-transform-async-to-promises/issues/37

**Comment by rpetrich** [comment]
Should be resolved by updating to 0.8.14 of babel-plugin-transform-async-to-promises

**Comment by jaredpalmer** [comment]
@rpetrich can you submit a PR?
